test/functional/rpc_psbt.py

```diff
@@ -820,6 +820,15 @@ def test_psbt_input_keys(psbt_input, keys):
             assert hash.hex() in res_input[preimage_key]
             assert_equal(res_input[preimage_key][hash.hex()], preimage.hex())
 
+        self.log.info("Test that combining PSBTs with different transactions fails")
+        tx = CTransaction()
+        tx.vin = [CTxIn(outpoint=COutPoint(hash=int('aa' * 32, 16), n=0), scriptSig=b"")]
+        tx.vout = [CTxOut(nValue=0, scriptPubKey=b"")]
+        psbt1 = PSBT(PSBTMap({PSBT_GLOBAL_UNSIGNED_TX: tx.serialize()}), [PSBTMap()], [PSBTMap()]).to_base64()
+        tx.vout[0].nValue += 1  # slightly modify tx
+        psbt2 = PSBT(PSBTMap({PSBT_GLOBAL_UNSIGNED_TX: tx.serialize()}), [PSBTMap()], [PSBTMap()]).to_base64()
+        assert_raises_rpc_error(-8, "PSBTs not compatible (different transactions)", self.nodes[0].combinepsbt, [psbt1, psbt2])
```

Comment:
Might as well check the naive `[psbt1, psbt1]` case to make sure that works

Reply:
Good idea, done.